src/components/ebay-date-textbox/component.ts

```diff
@@ -88,13 +89,23 @@ class DateTextbox extends Marko.Component<Input, State> {
                 : 2;
     }
 
-    handleInputChange(index: number, { value }: { value: string }) {
+    handleInputChange(index: number, { value }: TextboxEvent) {
         const valueDate = new Date(value);
         const iso = isNaN(valueDate.getTime()) ? null : dateArgToISO(valueDate);
         if (index === 0) {
-            this.state.firstSelected = iso;
+            if (this.state.secondSelected && iso > this.state.secondSelected) {
```

Comment:
I think there can be a cleaner way to do this. 
Lets first fix the issue with the date being swapped when typing manually. And then I will revisit this.